### ISSUE TITLE
GC overhead limit exceeded: accessing WebAPI/notifications

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/WebAPI#879]
### Expected Result
Launching Atlas should be able to load notifications without errors.

### Actual result
GC limit exceeded:
![image](https://user-images.githubusercontent.com/6818777/53525741-78fccf80-3ab0-11e9-8c7b-c52f74f5101d.png)

Notice that notifications is being continuously polled even before the prior request completes.

In the past, what we have done is we have a pollFor() method which is performed in a setTimeout, and then as long as polling is required, the last step of the setTimeoutCall is to call pollFor(). This means that the next pollFor() will only happen X seconds AFTER the previous one. But it appears that pollFor is being triggered every 10 seconds, and in that interval, you may not even receive a response from the first request, and then these things pile up.  I believe this pile-up is resulting in a GC error. However, I do not think the solution is to fix the polling, because we could get to this issue if we have 10 different clients polling for notifications.  Instead, we should understand why requesting notifications is so GC heavy (if it is infact the case that notifications is resulting in large requestss).

**Comment by chrisknoll** [comment]
I did some investigation:

When looking through the NotificationController (sigh) it pulls in the last 10 notifications via `findLast10`, which is implemented in NotificationServiceImpl (sigh) where it gets a page of jobExecutions.  Included in the job executions is the jobParamaters wich actually seems to contain the entirety of the execution design which could result in a several megabyte payload being returned from the database (just to get a list of executions).  

Later, these executions are being serialized down as a JobExecutionResource which includes all the job paramaters (again, this is all to just get a list of job executions for the notification service), which leads to a large payload being delivered to the client.

Prior to the refactoring, we would have small, streamlined DTOs (implemented as inner classes) that would have the minimum required amount of fields to satisfy a service request.  In the service layer, it would determine which slices of the entity objects need to be fetched, and only fetch those elements that are required.  Currently, it looks like there are a few 'one size fits all' dtos, which may be inappropriate for the targeted data elements that certain functions require.  Following 'form follows function' principle, we should have these DTO objects specifically designed to support the function that is being requested.

**Comment by anthonysena** [comment]
Just to add onto this issue, from an ATLAS standpoint the notifications area has changed. Here is how it used to look in 2.6:

![image](https://user-images.githubusercontent.com/12902366/53570341-3ed51180-3b34-11e9-82ec-761d91253cea.png)

And now in the master branch of Atlas:

![image](https://user-images.githubusercontent.com/12902366/53570404-6330ee00-3b34-11e9-84dd-d59116ac0eb9.png)

If you notice, the new notification area does not have a way to clear the job list (perhaps because this is all retrieved server-side now) but it also continues to poll for updates even when all of the jobs are completed which is also adding unnecessary overhead to WebAPI:

![image](https://user-images.githubusercontent.com/12902366/53570527-befb7700-3b34-11e9-9570-94173d794a9a.png)

**Comment by chrisknoll** [comment]
In our environment where we have 43 sources defined, the warmCache jobs start to fail after a few complete. errors include:

```
2019-02-28 23:48:42.739 ERROR taskExecutor-9 org.springframework.batch.core.job.AbstractJob -  - Encountered fatal error executing job
org.springframework.transaction.CannotCreateTransactionException: Could not open JPA EntityManager for transaction; nested exception is java.lang.OutOfMemoryError: GC overhead limit exceeded
	at org.springframework.orm.jpa.JpaTransactionManager.doBegin(JpaTransactionManager.java:431)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373)
```

Which leads to subsequent:
```
2019-02-28 23:50:25.581 WARN taskExecutor-9 org.hibernate.engine.jdbc.spi.SqlExceptionHelper -  - SQL Error: 0, SQLState: 08S01
2019-02-28 23:50:30.196 ERROR taskExecutor-9 org.hibernate.engine.jdbc.spi.SqlExceptionHelper -  - The TCP/IP connection to the host {xxxx}, port 1433 has failed. Error: "connect timed out. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.".
Mar 01, 2019 4:53:25 AM com.microsoft.sqlserver.jdbc.TDSReader throwInvalidTDS
SEVERE: ConnectionID:302 ClientConnectionId: 50a39578-eeb0-402f-9d6c-19880798d71a got unexpected value in TDS response at offset:267
```

Followed by:

```
[WARNING] 
java.lang.OutOfMemoryError: GC overhead limit exceeded

java.lang.OutOfMemoryError: GC overhead limit exceeded
[WARNING] 
java.lang.OutOfMemoryError: GC overhead limit exceeded

java.lang.OutOfMemoryError: GC overhead limit exceeded

java.lang.OutOfMemoryError: GC overhead limit exceeded
```

This is not confined to sql server connections:

```
]; Ran out of memory retrieving query results.; nested exception is org.postgresql.util.PSQLException: Ran out of memory retrieving query results.
```

**Comment by anthonysena** [comment]
Closed via #892 and OHDSI/Atlas#1389
